src/mkdocstrings/extension.py

```diff
@@ -231,17 +231,26 @@ def _process_block(
 
 class _PostProcessor(Treeprocessor):
     def run(self, root: Element) -> None:
+        self._remove_duplicated_headings(root)
+
+    def _remove_duplicated_headings(self, parent: Element) -> bool:
         carry_text = ""
-        for el in reversed(root):  # Reversed mainly for the ability to mutate during iteration.
+        found = False
+        for el in reversed(parent):  # Reversed mainly for the ability to mutate during iteration.
             if el.tag == "div" and el.get("class") == "mkdocstrings":
                 # Delete the duplicated headings along with their container, but keep the text (i.e. the actual HTML).
                 carry_text = (el.text or "") + carry_text
-                root.remove(el)
+                parent.remove(el)
+                found = True
             elif carry_text:
                 el.tail = (el.tail or "") + carry_text
                 carry_text = ""
+            elif self._remove_duplicated_headings(el):
+                found = True
+                break
```

Comment:
This has a wrong behavior: "If duplicated headings were found in a child element, stop scanning the rest of the document upwards". Why?

Reply:
I guess there's no specific "why" and the assumption is that duplicated headings appear in a single child element. Do you recommend to try and remove them from the whole subtree?

Reply:
Results from the child element are being propagated to the parent loop and it gets interrupted.